### PR TITLE
[RHOSO] Update proposed configs to new 18.0-fr1 branch

### PR DESCRIPTION
**Pull request by stuggi**
Updates the config for 18.0-fr1 branch and remove 18.0.0-proposed.
Where difference were between 18.0.0-proposed and main, update to
follow main as the feature branch was a rebase from main.

### CODE REVIEW

**Comment by stuggi** [comment]
/pj-rehearse pull-ci-openstack-k8s-operators-ironic-operator-18.0-fr1-images

**Comment by stuggi** [comment]
/pj-rehearse pull-ci-openstack-k8s-operators-ironic-operator-18.0-fr1-ironic-operator-build-deploy-kuttl

**Review by dprince — APPROVED** [review]

**Comment by stuggi** [comment]
```
  * could not run steps: step ironic-operator-build-deploy failed: failed to wait for the created cluster claim to become ready: timed out waiting for the condition 
```

**Comment by stuggi** [comment]
/pj-rehearse pull-ci-openstack-k8s-operators-ironic-operator-18.0-fr1-ironic-operator-build-deploy-kuttl

**Comment by stuggi** [comment]
/pj-rehearse pull-ci-openstack-k8s-operators-ironic-operator-18.0-fr1-ironic-operator-build-deploy

**Comment by stuggi** [comment]
```
error: build error: failed to pull image: After retrying 2 times, Pull image still failed due to error: initializing source docker://registry.access.redhat.com/ubi9/go-toolset:1.21: pinging container registry registry.access.redhat.com: Get "https://registry.access.redhat.com/v2/": dial tcp: lookup registry.access.redhat.com on 172.30.0.10:53: server misbehaving
```

**Comment by stuggi** [comment]
/pj-rehearse pull-ci-openstack-k8s-operators-ironic-operator-18.0-fr1-ironic-operator-build-deploy-kuttl

**Comment by stuggi** [comment]
/pj-rehearse pull-ci-openstack-k8s-operators-ironic-operator-18.0-fr1-ironic-operator-build-deploy

**Comment by stuggi** [comment]
/pj-rehearse pull-ci-openstack-k8s-operators-ironic-operator-18.0-fr1-ironic-operator-build-deploy

**Review by abays — APPROVED** [review]
/lgtm

File: ci-operator/config/openstack-k8s-operators/openstack-k8s-operators-ci/openstack-k8s-operators-openstack-k8s-operators-ci-main.yaml [hunk 1]
```diff
@@ -73,7 +73,7 @@ images:
   inputs:
     rhel-9-golang-1.21:
       as:
-      - registry.ci.openshift.org/openshift/release:rhel-9-release-golang-1.20-openshift-4.14
+      - registry.ci.openshift.org/openshift/release:rhel-9-release-golang-1.21-openshift-4.16
```
Comment (viroel) [inline ci-operator/config/openstack-k8s-operators/openstack-k8s-operators-ci/openstack-k8s-operators-openstack-k8s-operators-ci-main.yaml:76]: I know that looks weird and confusing, but you should keep as `registry.ci.openshift.org/openshift/release:rhel-9-release-golang-1.20-openshift-4.14`, and the explanation is:
The build args is actually saying: in Dockerfile `images/build-root/Dockerfile.rhel-9.custom` [1] replace `registry.ci.openshift.org/openshift/release:rhel-9-release-golang-1.20-openshift-4.14` with `rhel-9-golang-1.21`.
And `rhel-9-golang-1.21` is defined above, in line 18.
[1] https://github.com/openstack-k8s-operators/openstack-k8s-operators-ci/blob/e45ea3a5485ce0e9ce024bff16e475f2fa60cd7c/images/build-root/Dockerfile.rhel-9.custom#L1C77-L1C86

just double cheking if image is correct:
```
podman run registry.ci.openshift.org/openstack-k8s-operators/tools:ci-build-root-golang-1.21-sdk-1.31 go version && operator-sdk version
go version go1.21.11 (Red Hat 1.21.11-1.el9_4) linux/amd64
operator-sdk version: "v1.31.0", commit: "e67da35ef4fff3e471a208904b2a142b27ae32b1", kubernetes version: "1.26.0", go version: "go1.19.11", GOOS: "linux", GOARCH: "amd64"
```

**Review by viroel — CHANGES_REQUESTED** [review]

File: ci-operator/config/openstack-k8s-operators/openstack-k8s-operators-ci/openstack-k8s-operators-openstack-k8s-operators-ci-main.yaml (hunk 1 above)
Comment (stuggi) [reply ci-operator/config/openstack-k8s-operators/openstack-k8s-operators-ci/openstack-k8s-operators-openstack-k8s-operators-ci-main.yaml]: reverted this change

**Comment by stuggi** [comment]
/pj-rehearse pull-ci-openstack-k8s-operators-tcib-18.0-fr1-images

**Comment by stuggi** [comment]
/pj-rehearse pull-ci-openstack-k8s-operators-openstack-baremetal-operator-18.0-fr1-images

**Comment by stuggi** [comment]
/pj-rehearse pull-ci-openstack-k8s-operators-nova-operator-18.0-fr1-images

**Comment by stuggi** [comment]
thank you @deepsm007 !

**Pull request merged**